SurfaceDuoSamples/src/main/java/com/microsoft/device/display/samples/duosamples/DuoSamplesActivity.kt

```diff
@@ -8,10 +8,14 @@ package com.microsoft.device.display.samples.duosamples
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.FoldableNavigation
 import com.microsoft.device.display.samples.duosamples.databinding.ActivityDuoSamplesBinding
+import com.microsoft.device.display.samples.duosamples.navigation.MainNavigator
+import com.microsoft.device.display.samples.duosamples.navigation.MainNavigatorHost
 
-class DuoSamplesActivity : AppCompatActivity() {
+class DuoSamplesActivity : AppCompatActivity(), MainNavigatorHost {
```

Comment:
Maybe you also need to override the behavior for pressing back so the samples list fragment will not be popped and leave the activity empty?

Reply:
I've tested it and the navigation works fine.
I think it would be necessary if I would have opened another fragment. But here I am opining a new activity.